store/src/main/java/org/apache/rocketmq/store/index/IndexService.java

```diff
@@ -313,14 +313,24 @@ public IndexFile getAndCreateLastIndexFile() {
 
         if (indexFile == null) {
             try {
-                String fileName =
-                    this.storePath + File.separator
-                        + UtilAll.timeMillisToHumanString(System.currentTimeMillis());
-                indexFile =
-                    new IndexFile(fileName, this.hashSlotNum, this.indexNum, lastUpdateEndPhyOffset,
-                        lastUpdateIndexTimestamp);
                 this.readWriteLock.writeLock().lock();
-                this.indexFileList.add(indexFile);
+
+                IndexFile tmp = null;
+                if (!this.indexFileList.isEmpty()) {
+                    tmp = this.indexFileList.get(this.indexFileList.size() - 1);
+                }
+
+                if (null == tmp || tmp.isWriteFull()) {
```

Comment:
should we update `prevIndexFile` when `tmp.isWriteFull()` is true ?

Reply:
tmp may null you' ll find

Reply:
yeah that's why we might want to separate these two conditions. If `tmp` is not null and isWriteFull, then it is likely that  another thread created this `tmp` index file, which is different from the previous `prevIndexFile`. Probably I'm wrong LOL

Reply:
o,maybe my mistake， i don't have the idea to change whole code，i just try to fix the bug without change the conditions